File: reimi/frontend/reimi_app/lib/domain/value_objects/occupation.dart
import 'package:freezed_annotation/freezed_annotation.dart';

enum Occupation {
  // 学生
  @JsonValue('UNIVERSITY_STUDENT')
  universityStudent,
  @JsonValue('GRADUATE_STUDENT')
  graduateStudent,
  @JsonValue('VOCATIONAL_STUDENT')
  vocationalStudent,
  @JsonValue('JUNIOR_COLLEGE_STUDENT')
  juniorCollegeStudent,
  @JsonValue('TECHNICAL_COLLEGE_STUDENT')
  technicalCollegeStudent,

  // 業界
  @JsonValue('DISTRIBUTION')
  distribution,
  @JsonValue('FOOD_INDUSTRY')
  foodIndustry,
  @JsonValue('PHARMACEUTICAL')
  pharmaceutical,
  @JsonValue('MEDICAL')
  medical,
  @JsonValue('WELFARE_CARE')
  welfareCare,

  // 医療専門職
  @JsonValue('DOCTOR')
  doctor,
  @JsonValue('NURSE')
  nurse,
  @JsonValue('PHARMACIST')
  pharmacist,

  // IT・通信
  @JsonValue('TELECOMMUNICATION')
  telecommunication,
  @JsonValue('WEB')
  web,
  @JsonValue('IT')
  it,
  @JsonValue('ENGINEER')
  engineer,
  @JsonValue('CREATOR')
  creator,
  @JsonValue('DESIGNER')
  designer,

  // 建築・金融
  @JsonValue('ARCHITECTURE_INTERIOR')
  architectureInterior,
  @JsonValue('FINANCE')
  finance,
  @JsonValue('INSURANCE')
  insurance,
  @JsonValue('CONSULTING')
  consulting,

  // メディア
  @JsonValue('MEDIA')
  media,
  @JsonValue('ADVERTISING')
  advertising,
  @JsonValue('PUBLISHING')
  publishing,

  // 教育・流通
  @JsonValue('EDUCATION')
  education,
  @JsonValue('RETAIL')
  retail,
  @JsonValue('TRANSPORTATION')
  transportation,

  // 娯楽・サービス
  @JsonValue('ENTERTAINMENT')
  entertainment,
  @JsonValue('TRAVEL')
  travel,
  @JsonValue('REAL_ESTATE')
  realEstate,

  // ビジネス
  @JsonValue('TRADING_COMPANY')
  tradingCompany,
  @JsonValue('MANUFACTURER')
  manufacturer,
  @JsonValue('RESEARCHER')
  researcher,

  // 企業属性
  @JsonValue('MAJOR_COMPANY')
  majorCompany,
  @JsonValue('FOREIGN_COMPANY')
  foreignCompany,
  @JsonValue('LISTED_COMPANY')
  listedCompany,

  // 経営・法務
  @JsonValue('EXECUTIVE')
  executive,
  @JsonValue('LEGAL')
  legal,
  @JsonValue('LAWYER')
  lawyer,
  @JsonValue('CPA')
  certifiedPublicAccountant,
  @JsonValue('TAX_ACCOUNTANT')
  taxAccountant,

  // 公共・自由業
  @JsonValue('FREELANCE')
  freelance,
  @JsonValue('PUBLIC_SERVANT')
  publicServant,
  @JsonValue('FIREFIGHTER')
  firefighter,
  @JsonValue('POLICE')
  police,
  @JsonValue('SELF_DEFENSE_FORCE')
  selfDefenseForce,

  // 特殊職
  @JsonValue('ATHLETE')
  athlete,
  @JsonValue('PILOT')
  pilot,
  @JsonValue('INVESTOR')
  investor,
  @JsonValue('CABIN_ATTENDANT')
  cabinAttendant,

  // 生活・接客
  @JsonValue('APPAREL')
  apparel,
  @JsonValue('BEAUTY')
  beauty,
  @JsonValue('BRIDAL')
  bridal,
  @JsonValue('CHEF_NUTRITIONIST')
  chefNutritionist,
  @JsonValue('CHILDCARE_WORKER')
  childcareWorker,
  @JsonValue('SERVICE')
  service,
  @JsonValue('SECRETARY')
  secretary,
  @JsonValue('ANNOUNCER')
  announcer,
  @JsonValue('RECEPTIONIST')
  receptionist,

  // 芸能・ネット
  @JsonValue('ENTERTAINER_MODEL')
  entertainerModel,
  @JsonValue('INFLUENCER')
  influencer,
  @JsonValue('YOUTUBER')
  youtuber,
  @JsonValue('PRO_GAMER')
  proGamer,

  // その他
  @JsonValue('COMPANY_EMPLOYEE')
  companyEmployee,
  @JsonValue('OFFICE_WORK')
  officeWork,
  @JsonValue('SELF_EMPLOYED')
  selfEmployed,
  @JsonValue('OTHER')
  other,
}

File: reimi/backend/reimi_app/src/main/java/com/reimi/reimi_app/domain/model/profile/Occupation.java
package com.reimi.reimi_app.domain.model.profile;

public enum Occupation {

    // 学生
    UNIVERSITY_STUDENT("大学生"),
    GRADUATE_STUDENT("大学院生"),
    VOCATIONAL_STUDENT("専門学生"),
    JUNIOR_COLLEGE_STUDENT("短大生"),
    TECHNICAL_COLLEGE_STUDENT("高専生"),

    // 業界
    DISTRIBUTION("流通"),
    FOOD_INDUSTRY("食品関係"),
    PHARMACEUTICAL("製薬"),
    MEDICAL("医療関係"),
    WELFARE_CARE("福祉・介護"),

    // 医療専門職
    DOCTOR("医師"),
    NURSE("看護師"),
    PHARMACIST("薬剤師"),

    // IT・通信
    TELECOMMUNICATION("通信"),
    WEB("WEB関係"),
    IT("IT関係"),
    ENGINEER("エンジニア"),
    CREATOR("クリエイター"),
    DESIGNER("デザイナー"),

    // 建築・金融
    ARCHITECTURE_INTERIOR("建築・インテリア"),
    FINANCE("金融"),
    INSURANCE("保険"),
    CONSULTING("コンサル"),

    // メディア
    MASS_MEDIA("マスコミ"),
    ADVERTISING("広告"),
    PUBLISHING("出版"),

    // 教育・流通
    EDUCATION("教育関係"),
    RETAIL("小売"),
    TRANSPORTATION("輸送・交通"),

    // 娯楽・サービス
    ENTERTAINMENT("エンターテイメント"),
    TRAVEL("旅行関係"),
    REAL_ESTATE("不動産"),

    // ビジネス
    TRADING_COMPANY("商社"),
    MANUFACTURER("メーカー"),
    RESEARCHER("研究職"),

    // 企業属性
    MAJOR_COMPANY("大手企業"),
    FOREIGN_COMPANY("外資企業"),
    LISTED_COMPANY("上場企業"),

    // 経営・法務
    EXECUTIVE("経営者・役員"),
    LEGAL("法務関係"),
    LAWYER("弁護士"),
    CPA("公認会計士"),
    TAX_ACCOUNTANT("税理士"),

    // 公共・自由業
    FREELANCE("自由業"),
    PUBLIC_SERVANT("公務員"),
    FIREFIGHTER("消防士"),
    POLICE("警察官"),
    SELF_DEFENSE_FORCE("自衛隊"),

    // 特殊職
    ATHLETE("スポーツ選手"),
    PILOT("パイロット"),
    INVESTOR("投資家"),
    CABIN_ATTENDANT("キャビンアテンダント"),

    // 生活・接客
    APPAREL("アパレル"),
    BEAUTY("美容関係"),
    BRIDAL("ブライダル"),
    CHEF_NUTRITIONIST("調理師・栄養士"),
    CHILDCARE_WORKER("保育士"),
    SERVICE("サービス業"),
    SECRETARY("秘書"),
    ANNOUNCER("アナウンサー"),
    RECEPTIONIST("受付"),

    // 芸能・ネット
    ENTERTAINER_MODEL("芸能・モデル"),
    INFLUENCER("インフルエンサー"),
    YOUTUBER("YouTuber"),
    PRO_GAMER("プロゲーマー"),

    // その他
    COMPANY_EMPLOYEE("会社員"),
    OFFICE_WORK("事務"),
    SELF_EMPLOYED("自営業"),
    OTHER("その他");

    private final String label;

    Occupation(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}
